07.ls_object/lib/file_data.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+class FileData
+  attr_reader :base_name
+
+  def initialize(path)
+    @base_name = File.basename(path)
```

Comment:
```suggestion
    @path = path
```

pathの方が情報量が多いのでこっちを保持しましょう

Reply:
「pathの方が情報量が多く、そのことで起こる不具合もないので、新たな機能を追加したいとなったときに利用しやすい」という考え方でよろしいでしょうか。
コードとしてもより簡潔になった気がします。

Reply:
> 「pathの方が情報量が多く、そのことで起こる不具合もないので、新たな機能を追加したいとなったときに利用しやすい」という考え方でよろしいでしょうか。

合ってます！

> インスタンス変数としては@pathでpath情報を保持しつつ、ロジックには新たに作成したbase_nameメソッドを利用するようにしました。

この考え方でOKです 👍